tests/modules/test_cloud.py

```diff
@@ -39,50 +39,104 @@ def assert_tensor_list_diff(a, b):
+        with mock.patch("ablator.modules.storage.cloud.GcpConfig.list_bucket",mock_list_bucket_empty):
+            with mock.patch("socket.gethostname", return_value="localhost"):
+                assert_error_msg(
+                    lambda: GcpConfig(bucket=bucket),
+                    f"Can only use GcpConfig from Google Cloud Server. Consider switching to RemoteConfig.",
+                )
+    
+    
+    with mock.patch("socket.gethostname", return_value="gcp-machine1"):
+        with mock.patch("socket.gethostbyname",return_value="111.111.111.111"):
+            with mock.patch("ablator.modules.storage.cloud.GcpConfig._find_gcp_nodes",return_value=[{"networkInterfaces":[{"networkIP":"111.111.111.111"}]}]):
+                with mock.patch("ablator.modules.storage.cloud.GcpConfig.list_bucket",mock_list_bucket_non_empty):
+                    with mock.patch("ablator.modules.storage.cloud.GcpConfig.rsync_up",mock_rsync_up):
+                        cfg = GcpConfig(bucket=bucket)
+                        files = cfg.list_bucket()
+                        original_tensors = write_rand_tensors(tmp_path)
+                        cfg.rsync_up(tmp_path, rand_folder)
+
+    
+    with mock.patch("ablator.modules.storage.cloud.GcpConfig.list_bucket",return_value={"gs://iordanis/bucket1","gs://iordanis/bucket2",f"gs://{Path(cfg.bucket) / rand_folder}/"}):
+        new_files = cfg.list_bucket()
     rand_destination = f"gs://{Path(cfg.bucket) / rand_folder}/"
     assert set(new_files).difference(files) == {rand_destination}
-    uploaded_files = cfg.list_bucket(rand_folder + "/" + tmp_path.name)
+    uploaded_files=None
+    with mock.patch("ablator.modules.storage.cloud.GcpConfig.list_bucket",return_value={f"{rand_destination}/{tmp_path.name}/tenosr1",f"{rand_destination}/{tmp_path.name}//tenosr2"}):
+        uploaded_files = cfg.list_bucket(rand_folder + "/" + tmp_path.name)
     assert len(uploaded_files) == 2
     # Replace original tensors
     new_tensors = write_rand_tensors(tmp_path)
     assert_tensor_list_diff(original_tensors, new_tensors)
     loaded_tensors = load_rand_tensors(tmp_path=tmp_path)
     assert_tensor_list_eq(loaded_tensors, new_tensors)
     # Update the local tensors with the original tensors from gcp
-    cfg.rsync_down(rand_folder, tmp_path, verbose=False)
+    
+    def mock_sync_down_new_file(self,remote_path: str,local_path: Path,logger: FileLogger | None = None,verbose=True):
+        for i in range(2):
+            torch.save(original_tensors[i],tmp_path.joinpath(f"t_{i}.pt"))
+    with mock.patch("ablator.modules.storage.cloud.GcpConfig.rsync_down",mock_sync_down_new_file):
+        cfg.rsync_down(rand_folder, tmp_path, verbose=False)
     loaded_tensors = load_rand_tensors(tmp_path=tmp_path)
     assert_tensor_list_eq(loaded_tensors, original_tensors)
 
     # Update a mock node from gcp
-    hostname = socket.gethostname()
+    with mock.patch("socket.gethostname", return_value="localhost"):
```

Comment:
this is unnecessary you can set hostname = "localhost"

Reply:
done.